### PR TITLE
Add preference to ignore middleware

### PR DESCRIPTION
**Pull request by patgmac**
This PR adds the preference IgnoreMiddleware. This could be helpful when moving away from using middleware because of a chicken & egg scenario where your new SoftwareRepoURL no longer needs middleware.

### CODE REVIEW

**Comment by gregneagle** [comment]
If merged, this probably wouldn’t make it into a release any time soon. If one had an immediate need, the middleware itself could check for the/a preference and return having done nothing at all…

**Comment by patgmac** [comment]
Understood. I do have other ideas on how to handle this, but they involved using other tools (WS1 or Tanium). I do like the idea of handling this in a new revision of the middleware itself to check for that preference. I may go that route if needed. 

I thought this might be a good idea because of projects such as https://github.com/jc0b/munki-gcs-redirector which is a GCP cloud run service that does not need middleware and some of us previously needed to use middleware to talk to GCP (or other cloud providers).

File: code/client/munkilib/fetch.py [hunk 1]
```diff
@@ -118,7 +118,11 @@ def import_middleware():
 
 
 middleware = None
-import_middleware()
+ignore_middleware = prefs.pref('IgnoreMiddleware')
+if ignore_middleware is not True:
```
Comment (gregneagle) [inline code/client/munkilib/fetch.py:122]: Arguably more Pythonic to say
```
if not ignore_middleware:
```

**Comment by gregneagle** [comment]
Still trying to decide if the pref should be "ignoreMiddleware" with a default of False, or "useMiddleware" with a default of True...

**Comment by gregneagle** [comment]
Crud. Forgot this was here. I'll merge it now for a future (post 6.4.1) release.

**Pull request merged**